### PR TITLE
[Relay] fetch voting & waiting for tx reports

### LINKED ISSUE #374: [UST-119][Relay] Fetch Voting Report API
**Is your feature request related to a problem? Please describe.**
### create fetch report API (GET /api/reports)
should fetch reports which meet below conditions for client side to display to active user to vote.

- reported epoch = current epoch - 1
- **tie vote situation(agree vote = disagree vote)** and **reported epoch < current epoch**
### test case
- fetch all reports whose epoch = current epoch - 1
- fetch all reports whose status is tie vote and reported epoch < current epoch
- reports whose reported epoch is at current epoch shouldn't be fetched.

### LINKED ISSUE #375: [UST-170][Relay] Fetch WaitingForTx Report API
**Is your feature request related to a problem? Please describe.**

### create an api to fetch all reports whose status is waiting for transaction
The api would be like `GET /api/reports?status=2`, client side calls this api to claim reputation for 檢舉者 / 被檢舉者 in the background. Should fetch reports which meet below conditions
- Report with status =  2 (Waiting for Tx)
### test case
- get reports whose status is waiting for transaction only
- throw error if status is invalid (ex. status=888)

### PR DESCRIPTION
**Pull request by chentihe**
## Summary

create an api for client side to fetch `Voting` and `WaitingForTransaction` reports filtered by status(0, 1)

`/api/report?status=0` : fetching Voting reports
`/api/report?status=1` : fetching WaitingForTransaction reports

## Linked Issue

close #374 
close #375 

## Details

this api assume that all reports are handled well by synchronizer, so the relayer doesn't check the adjudication result.

## Tests

1. should fetch report whose reportEpoch is equal to currentEpoch - 1
2. should fail to fetch report with wrong query status or without status query params
3. should fetch report whose adjudication result is tie
4. should fetch report whose adjudication count is less than 5
5. should fetch report whose status is waiting for transaction

## Verification Steps

`yarn relay test`

## Checklist

-   [x] All new and existing tests pass
-   [x] I have updated the documentation (if applicable)
-   [x] My changes do not introduce new security vulnerabilities

### CODE REVIEW

File: packages/relay/test/report.test.ts [hunk 1]
```diff
@@ -255,4 +261,131 @@ describe('POST /api/report', function () {
                 expect(res.body.error).to.be.equal('Invalid postId')
             })
     })
+
+    it('should fetch report whose reportEpoch is equal to currentEpoch - 1', async function () {
```
Comment (ryanycw) [inline packages/relay/test/report.test.ts:265]: Can we add one more test here is that, reportEpoch is equal to currentEpoch than empty list will return?

File: packages/relay/src/services/ReportService.ts [hunk 2]
```diff
@@ -78,20 +81,65 @@ export class ReportService {
...
+                        ],
+                    },
+                })
+                break
+            default:
+                throw InvalidReportStatusError
+        }
+
+        return reports
+    }
```
Comment (ryanycw) [inline packages/relay/src/services/ReportService.ts:142]:
```suggestion
        const statusConditions = {
            // VOTEING reports meet below condition
            // 1. reportEpoch = current Epoch - 1
            // 2. the result of adjudication is tie, should vote again
            // p.s. synchronizer would handle the adjudication result,
            // we can assume all reports whose status is VOTING are already handled by synchronizer
            [ReportStatus.VOTING]: {
                AND: [
                    { reportEpoch: { lt: epoch } },
                    { status: ReportStatus.VOTING },
                ],
            },
            // WAITING_FOR_TRANSACTION is for client side to claim reputation use
            // reports whose report epoch is equal to current epoch are pending reports
            // the status should be always VOTING, so the where clause don't search
            // current epoch
            [ReportStatus.WAITING_FOR_TRANSACTION]: {
                AND: [
                    { reportEpoch: { lt: epoch } },
                    { status: ReportStatus.WAITING_FOR_TRANSACTION },
                ],
            }
        }
    
        const condition = statusConditions[status];
        if (!condition) {
            throw InvalidReportStatusError
        }
    
        return await db.findMany('ReportHistory', { where: condition })
```

Comment (ryanycw) [reply packages/relay/src/services/ReportService.ts:142]: Would prefer this to make it cleaner

**Review by ryanycw — CHANGES_REQUESTED** [review]
All good, some minor feedback that would be better to include.

File: packages/relay/test/report.test.ts (hunk 1 above)
Comment (chentihe) [reply packages/relay/test/report.test.ts:272]: 👍🏻

**Review by ryanycw — APPROVED** [review]
